flixel/addons/editors/ogmo/FlxOgmo3Loader.hx

```diff
@@ -0,0 +1,514 @@
+		var out:Map<String, Array<FlxPoint>> = [];
+		switch gridLayer.arrayMode
+		{
+			case 0:
+				for (i in 0...gridLayer.grid.length)
+				{
+					if (!out.exists(gridLayer.grid[i]))
+						out.set(gridLayer.grid[i], []);
+					out[gridLayer.grid[i]].push(FlxPoint.get((i % gridLayer.gridCellsX) * gridLayer.gridCellWidth,
+						(i / gridLayer.gridCellsX).floor() * gridLayer.gridCellHeight));
+				}
+			case 1:
+				for (j in 0...gridLayer.grid2D.length)
+					for (i in 0...gridLayer.grid2D[j].length)
+					{
+						if (!out.exists(gridLayer.grid2D[j][i]))
+							out.set(gridLayer.grid2D[j][i], []);
+						out[gridLayer.grid2D[j][i]].push(FlxPoint.get(i * gridLayer.gridCellWidth, j * gridLayer.gridCellHeight));
+					}
+		}
+		return out;
+	}
+
+	/**
+	 * Parse every entity in the specified layer and call a function that will spawn game objects based on its entity data.
+	 * Here's an example that reads the position of an object:
+	 *
+	 * ```haxe
+	 *	function loadEntity(entity:EntityData)
+	 *	{
+	 * 		switch (entity.name)
+	 *		{
+	 *			case "player":
+	 *				player.x = entity.x;
+	 *				player.y = entity.y;
+	 *				player.custom_value = entity.values.custom_value;
+	 *			default:
+	 *				throw 'Unrecognized actor type ${entity.name}';
+	 *		}
+	 *	}
+	 * ```
+	 *
+	 * @param	EntityLoadCallback		A function with the signature `(name:String, data:Xml):Void` and spawns entities based on their name.
+	 * @param	EntityLayer				The name of the layer the entities are stored in Ogmo editor. Usually `"entities"` or `"actors"`.
+	 */
+	public function loadEntities(EntityLoadCallback:EntityData->Void, EntityLayer:String = "entities"):Void
+	{
+		for (entity in level.getEntityLayer(EntityLayer).entities)
+			EntityLoadCallback(entity);
+	}
+
+	/**
+	 * Loads every decal in a decal layer into a FlxGroup.
+	 *
+	 * IMPORTANT: All decals must be included in one directory!
+	 *
+	 * @param DecalLayer	The name of the layer the decals are stored in Ogmo editor. Usually `"decals"`.
+	 * @param decalsPath 	The path to the directory in which your decal assets are stored.
+	 * @return FlxGroup
+	 */
+	public function loadDecals(DecalLayer:String = 'decals', decalsPath:String):FlxGroup
+	{
+		if (decalsPath.charAt(decalsPath.length - 1) != '/')
+			decalsPath += '/';
+		var g = new FlxGroup();
+		for (decal in level.getDecalLayer(DecalLayer).decals)
+		{
+			var s = new FlxSprite(decal.x, decal.y, decalsPath + decal.texture);
+			s.offset.set(s.width / 2, s.height / 2);
+			if (decal.scaleX != null)
+				s.scale.x = decal.scaleX;
+			if (decal.scaleY != null)
+				s.scale.y = decal.scaleY;
+			if (decal.rotation != null)
+				s.angle = project.anglesRadians ? decal.rotation * 180 / Math.PI : decal.rotation;
+			g.add(s);
+		}
+		return g;
+	}
+
+	/**
+	 * Parse OGMO Editor level .json text
+	 * @param json
+	 * @return LevelData
+	 */
+	static function parseLevelJSON(json:String):LevelData
+	{
+		return cast json.parse();
```

Comment:
Not a huge fan of static extensions for `haxe.Json` either. :P None of these classes were really designed for static extension.

Reply:
👍